src/pybind/mgr/dashboard/frontend/src/app/shared/services/formatter.service.ts

```diff
@@ -59,42 +37,15 @@ export class FormatterService {
    */
   toBytes(value: string): number | null {
     const base = 1024;
-    const units = {
-      'b': 1,
-      'k': Math.pow(base, 1),
-      'kb': Math.pow(base, 1),
-      'kib': Math.pow(base, 1),
-      'm': Math.pow(base, 2),
-      'mb': Math.pow(base, 2),
-      'mib': Math.pow(base, 2),
-      'g': Math.pow(base, 3),
-      'gb': Math.pow(base, 3),
-      'gib': Math.pow(base, 3),
-      't': Math.pow(base, 4),
-      'tb': Math.pow(base, 4),
-      'tib': Math.pow(base, 4),
-      'p': Math.pow(base, 5),
-      'pb': Math.pow(base, 5),
-      'pib': Math.pow(base, 5),
-      'e': Math.pow(base, 6),
-      'eb': Math.pow(base, 6),
-      'eib': Math.pow(base, 6),
-      'z': Math.pow(base, 7),
-      'zb': Math.pow(base, 7),
-      'zib': Math.pow(base, 7),
-      'y': Math.pow(base, 8),
-      'yb': Math.pow(base, 8),
-      'yib': Math.pow(base, 8)
-    };
-    const m = RegExp('^(\\d+)\\s*(B|K(B|iB)?|M(B|iB)?|G(B|iB)?|T(B|iB)?|P(B|iB)?|' +
-      'E(B|iB)?|Z(B|iB)?|Y(B|iB)?)?$', 'i').exec(value);
+    const units = ['b', 'k', 'm', 'g', 't', 'p', 'e', 'z', 'y'];
+    const m = RegExp('^(\\d+(\.\\d+)?) ?(\[' + units.join('') + '\]i?b?)?$', 'i').exec(value);
```

Comment:
This regex validates a input like ``10Gi`` as valid what is not correct.

Reply:
You need to use ``RegExp('^(\\d+(\.\\d+)?) ?(\[' + units.join('') + '\](b|ib)?)?$', 'i')``

Reply:
`10Gi` would be converted into `10GiB`  -  but i will change that